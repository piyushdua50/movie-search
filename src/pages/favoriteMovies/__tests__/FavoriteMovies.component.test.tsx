import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import FavoriteMovies from "../FavoriteMovies.component";
test("render the FavoriteMovies screen correctly", () => {
  const initialState = {
    movies: {
      movies: [
        {
          Title: "Kuch Tum Kaho Kuch Hum Kahein",
          Year: "2002",
          imdbID: "tt0327004",
          Type: "movie",
          Poster:
            "https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg",
        },
      ],
      favoriteMovies: [
        {
          Title: "Kuch Tum Kaho Kuch Hum Kahein",
          Year: "2002",
          imdbID: "tt0327004",
          Type: "movie",
          Poster:
            "https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg",
        },
      ],
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <Routes>
          <Route path="/" element={<FavoriteMovies />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
