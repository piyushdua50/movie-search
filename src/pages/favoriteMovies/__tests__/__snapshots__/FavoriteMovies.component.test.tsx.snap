// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the FavoriteMovies screen correctly 1`] = `
<div>
  <div
    class="favoriteContainer"
  >
    <div
      class="headerContainer"
    >
      <h1
        class="headerTitle"
      >
        Home
      </h1>
      <h1
        class="headerTitle"
      >
        Favorites
      </h1>
    </div>
    <div
      class="movieListContainer"
    >
      <div
        class="movieContainer"
      >
        <img
          alt="Kuch Tum Kaho Kuch Hum Kahein"
          class="imgContainer"
          src="https://m.media-amazon.com/images/M/MV5BNGVhMjBhYmQtMDFmZC00Mjc2LWIwZTYtMmJiY2ZkNjM4N2FlXkEyXkFqcGdeQXVyNDUzOTQ5MjY@._V1_SX300.jpg"
        />
        <span
          class="movieNameContainer"
        >
          Kuch Tum Kaho Kuch Hum Kahein
        </span>
        <div
          class="movieInfoContainer"
        >
          <span
            class="movieInfo"
          >
            Year
            : 
            2002
          </span>
          <span
            class="movieInfo"
          >
            Type
            : 
            movie
          </span>
        </div>
        <div
          class="movieFavoriteInfo"
          role="button"
        >
          Favorite
        </div>
      </div>
    </div>
  </div>
</div>
`;
